File: src/typescript/src/pipeline_services/examples/text_steps.ts
export async function strip(text_value: unknown): Promise<string> {
  const text_string = String(text_value);
  return text_string.trim();
}

export async function normalize_whitespace(text_value: unknown): Promise<string> {
  const text_string = String(text_value);
  const normalized_value = text_string.replace(/\s+/g, " ").trim();
  return normalized_value;
}

export async function to_lower(text_value: unknown): Promise<string> {
  const text_string = String(text_value);
  return text_string.toLowerCase();
}

export async function append_marker(text_value: unknown): Promise<string> {
  const text_string = String(text_value);
  return text_string + "|";
}

File: src/Python/pipeline_services/examples/text_steps.py
from __future__ import annotations

import re


def strip(text_value: str) -> str:
    text_string = str(text_value)
    return text_string.strip()


def normalize_whitespace(text_value: str) -> str:
    text_string = str(text_value)
    normalized_value = re.sub(r"\s+", " ", text_string).strip()
    return normalized_value


def to_lower(text_value: str) -> str:
    text_string = str(text_value)
    return text_string.lower()


def append_marker(text_value: str) -> str:
    text_string = str(text_value)
    return text_string + "|"

